portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -54,13 +57,20 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws S
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("comment");
-    PreparedQuery results = datastore.prepare(query);
-    ArrayList<String> comments = new ArrayList<>();
-    for (Entity entity : results.asIterable()) {
-      comments.add((String)entity.getProperty("value"));
+    String maxCommentsString = request.getParameter("max_comments");
+    if (maxCommentsString != null) {
+      Integer maxComments = Integer.parseInt(maxCommentsString);
+      Query query = new Query(COMMENT_KIND);
+      List<Entity> results = datastore.prepare(query).asList(FetchOptions.Builder.withLimit(maxComments));
+      String[] comments = results
```

Comment:
Nice usage of the stream APIs. If I were using this in prod I might want to do a sanity check on any memory/CPU increase, but the reduced verbosity is very nice to have.